sdk/include/platform/arty-a7/platform-entropy.hh

```diff
@@ -0,0 +1,65 @@
+#pragma once
+#include "compartment-macros.h"
+#include <platform/concepts/entropy.h>
+#include <ds/xoroshiro.h>
+#include <riscvreg.h>
+#include <interrupt.h>
+#include <fail-simulator-on-error.h>
+
+DECLARE_AND_DEFINE_INTERRUPT_CAPABILITY(                               
+  RevokerInterruptEntropy, InterruptName::EthernetReceiveInterrupt, true, false)                                         
+
+
+/**
+ * A simple entropy source.  This wraps a few weak entropy sources to seed a
+ * PRNG.  It is absolutely not secure and should not be used for anything that
+ * depends on cryptographically secure random numbers!  Unfortunately, there is
+ * nothing on the Arty A7 instantiation of CHERIoT SAFE that can be used as a
+ * secure entropy source.
```

Comment:
Huh!  That's kind of surprising.  Maybe something like https://dl.acm.org/doi/10.1145/3610295 is worth adding to the FPGA image?  But either way, this looks fine for testing and demo purposes.

Reply:
There are a bunch of ways you can do entropy sources (free-running ring oscillators are a favourite) on FPGAs.  The problem is not that it's impossible, it's that no one has done it for CHERIoT SAFE.